crates/symbolicator/src/endpoints/symbolicate_jvm.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+
+use axum::{extract, Json};
+use serde::{Deserialize, Serialize};
+use symbolicator_proguard::interface::{
+    JvmException, JvmModule, JvmStacktrace, SymbolicateJvmStacktraces,
+};
+use symbolicator_sources::SourceConfig;
+
+use crate::service::{RequestService, SymbolicationResponse};
+use crate::utils::sentry::ConfigureScope;
+
+use crate::endpoints::symbolicate::SymbolicationRequestQueryParams;
+use crate::endpoints::ResponseError;
+
+#[derive(Serialize, Deserialize)]
+pub struct JvmSymbolicationRequestBody {
+    pub sources: Arc<[SourceConfig]>,
+    #[serde(default)]
+    pub exceptions: Vec<JvmException>,
+    #[serde(default)]
+    pub stacktraces: Vec<JvmStacktrace>,
+    #[serde(default)]
+    pub modules: Vec<JvmModule>,
+    #[serde(default)]
+    pub release_package: Option<String>,
+    #[serde(default)]
+    pub options: JvmRequestOptions,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct JvmRequestOptions {
+    /// Whether to apply source context for the stack frames.
+    #[serde(default = "default_apply_source_context")]
+    pub apply_source_context: bool,
+}
+
+fn default_apply_source_context() -> bool {
+    true
+}
+
+impl Default for JvmRequestOptions {
```

Comment:
Since line 34 has a default value, does this `impl` required? Can we just add `Default` to line 31 as a trait, and remove this function?

Reply:
I think the difference is that the `Default` impl covers the case of `options` not being present at all in the request, whereas the `default_apply_source_context` function covers the case of `options` being present, but missing `apply_source_context`. Not sure if that's a concern in practice, but we might as well cover our bases.

Reply:
Ah, makes sense. Thank you!